### PR TITLE
feat: switch from abandoned redlock library to @sesamecare-oss/redlock

### PR DESCRIPTION
**Pull request by dotslashx**
Hi, first of all, thanks for this module.

The original [node-redlock](https://github.com/mike-marcacci/node-redlock) library seems to be abandoned and in a broken state. I've created this pull request so you can switch to an alternative [@sesamecare-oss/redlock](https://github.com/sesamecare/redlock).

Here's some background of the issues that made me create this PR:
- https://github.com/mike-marcacci/node-redlock/issues/160
- https://github.com/mike-marcacci/node-redlock/issues/290
- https://github.com/mike-marcacci/node-redlock/issues/173

In short, the original redlock library, as of the v5 release cannot be used in any modern NodeJS project, because of the Node16/NodeNext module resolution. This is already fixed upstream but never made it to the release and it doesn't look like it will happen anytime soon.

There's only one drawback of this PR i know of - loosing the ability to provide custom acquire/extend/release scripts, but IMHO this is unlikely critical for this library.

### CODE REVIEW

**Comment by anchan828** [comment]
Hi @dotslashx, thanks for creating the PR.
I will carefully read this PR and the issue you shared within the next few days and respond accordingly. For now, switching packages might be a potential risk, so I may publishing a new package (e.g., @anchan828/nest-sesamecare-redlock 🤔).

**Comment by anchan828** [comment]
I released [@anchan828/nest-sesamecare-redlock](https://www.npmjs.com/package/@anchan828/nest-sesamecare-redlock) package. I've decided to provide it as a separate package after all. Please use it instead.

It is true that many people are facing difficulties, but since this issue has not yet been resolved, I felt that switching packages in a wrapper library like this repository is premature.

- Even if switch at this point,[@sesamecare-oss/redlock](https://github.com/sesamecare/redlock) may not necessarily fix bugs or develop further. Therefore, there is a possibility that you will have to switch packages again.
- There is a possibility that people using `@anchan828/nest-redlock` package are also using the node-redlock package in the same project. If this is the case, users will have to install two packages.

Since this repository is a thin wrapper, it will be easy to maintain even if we provide a similar package temporarily.

@dotslashx I was unable to merge this PR, but thank you for creating and proposing it.
